aten/src/ATen/native/cuda/DistributionTemplates.h

```diff
@@ -401,6 +401,34 @@ struct RandomKernel {
   }
 };
 
+// ==================================================== Macros ========================================================
+
+#define UNIFORM_AND_TRANSFORM(scalar_t, accscalar_t, curand4_engine_calls, iter, gen, transform)                      \
```

Comment:
Is there a reason why these macros can't just be templated functions?

Reply:
no, I will fix it